Header header
bool im_true
float64[4] means